lib/extension-es6.js

```diff
@@ -7,7 +7,8 @@ function es6(loader) {
 
   var transpiler, transpilerModule, transpilerRuntimeModule, transpilerRuntimeGlobal;
 
-  var isBrowser = typeof window != 'undefined';
+  var isBrowser = typeof window !== 'undefined' && !!window.Window && window instanceof window.Window;
+  var isWorker = typeof self !== 'undefined' && !!self.WorkerGlobalScope && self instanceof WorkerGlobalScope;
```

Comment:
+1 for `autoloadTranspiler`

I figured that checking for the type of `window` is not sufficient since anyone could "accidently" set it. In my opinion `instanceof` should be checked as well. Same thing for the web worker, one could create a global property `WorkerGlobalScope`.

In [polyfill-wrapper-end.js](https://github.com/systemjs/systemjs/blob/master/lib/polyfill-wrapper-end.js#L26) the `instanceof`check is also done.

Actually it should look like this because if `WorkerGlobalScope` is not a function you get a `TypeError`.

```js
var isBrowser = typeof window !== 'undefined' && typeof Window === 'function' && window instanceof Window;
var isWorker = typeof self !== 'undefined' &&  typeof WorkerGlobalScope === 'function' && self instanceof WorkerGlobalScope;
```

Since there are now to occurrences of these checks it would be better to reuse them.